RecoEgamma/EgammaElectronAlgos/src/ElectronSeedGenerator.cc

```diff
@@ -17,188 +17,112 @@
-  };
-
-  Count stcount;
-}  // namespace
-#endif
-
 ElectronSeedGenerator::ElectronSeedGenerator(const edm::ParameterSet &pset, const ElectronSeedGenerator::Tokens &ts)
-    : dynamicphiroad_(pset.getParameter<bool>("dynamicPhiRoad")),
-      fromTrackerSeeds_(pset.getParameter<bool>("fromTrackerSeeds")),
-      useRecoVertex_(false),
+    : dynamicPhiRoad_(pset.getParameter<bool>("dynamicPhiRoad")),
       verticesTag_(ts.token_vtx),
       beamSpotTag_(ts.token_bs),
-      lowPtThreshold_(pset.getParameter<double>("LowPtThreshold")),
-      highPtThreshold_(pset.getParameter<double>("HighPtThreshold")),
+      lowPtThresh_(pset.getParameter<double>("LowPtThreshold")),
+      highPtThresh_(pset.getParameter<double>("HighPtThreshold")),
       nSigmasDeltaZ1_(pset.getParameter<double>("nSigmasDeltaZ1")),
-      deltaZ1WithVertex_(0.5),
+      deltaZ1WithVertex_(pset.getParameter<double>("deltaZ1WithVertex")),
       sizeWindowENeg_(pset.getParameter<double>("SizeWindowENeg")),
       deltaPhi1Low_(pset.getParameter<double>("DeltaPhi1Low")),
       deltaPhi1High_(pset.getParameter<double>("DeltaPhi1High")),
-      deltaPhi1Coef1_(0.),
-      deltaPhi1Coef2_(0.),
-      myMatchEle(nullptr),
-      myMatchPos(nullptr),
-      thePropagator(nullptr),
-      theMeasurementTracker(nullptr),
-      theMeasurementTrackerEventTag(ts.token_measTrkEvt),
-      theSetup(nullptr),
-      cacheIDMagField_(0),
-      /*cacheIDGeom_(0),*/ cacheIDNavSchool_(0),
-      cacheIDCkfComp_(0),
-      cacheIDTrkGeom_(0) {
-  // so that deltaPhi1 = deltaPhi1Coef1_ + deltaPhi1Coef2_/clusterEnergyT
-  if (dynamicphiroad_) {
-    deltaPhi1Coef2_ = (deltaPhi1Low_ - deltaPhi1High_) / (1. / lowPtThreshold_ - 1. / highPtThreshold_);
-    deltaPhi1Coef1_ = deltaPhi1Low_ - deltaPhi1Coef2_ / lowPtThreshold_;
+      // so that deltaPhi1 = dPhi1Coef1_ + dPhi1Coef2_/clusterEnergyT
+      dPhi1Coef2_(dynamicPhiRoad_ ? (deltaPhi1Low_ - deltaPhi1High_) / (1. / lowPtThresh_ - 1. / highPtThresh_) : 0.),
+      dPhi1Coef1_(dynamicPhiRoad_ ? deltaPhi1Low_ - dPhi1Coef2_ / lowPtThresh_ : 0.),
+      propagator_(nullptr),
+      measurementTracker_(nullptr),
+      measurementTrackerEventTag_(ts.token_measTrkEvt),
+      setup_(nullptr),
+      measurementTrackerName_(pset.getParameter<std::string>("measurementTrackerName")),
+      // use of reco vertex
+      useRecoVertex_(pset.getParameter<bool>("useRecoVertex")),
+      // new B/F configurables
+      deltaPhi2B_(pset.getParameter<double>("DeltaPhi2B")),
+      deltaPhi2F_(pset.getParameter<double>("DeltaPhi2F")),
+      phiMin2B_(pset.getParameter<double>("PhiMin2B")),
+      phiMin2F_(pset.getParameter<double>("PhiMin2F")),
+      phiMax2B_(pset.getParameter<double>("PhiMax2B")),
+      phiMax2F_(pset.getParameter<double>("PhiMax2F")),
+      electronMatcher_(pset.getParameter<double>("ePhiMin1"),
+                       pset.getParameter<double>("ePhiMax1"),
+                       phiMin2B_,
```

Comment:
would std::bind be useful here to be less repetitive in electronMatcher_ and positronMatcher_?

Reply:
I thought of that too, but concluded it's a waste of time as it is probably only an intermediate solution. There is nothing that makes the electron and positron matcher really different except for a sign flip in the phi-window. IMHO one could just adapt the matcher class a bit and only use one instance after all. I'm afraid doing that would blow up this PR, however.